### ISSUE TITLE
no save_weights function

### ISSUE THREAD

**Issue by SidHard** [eriklindernoren/PyTorch-YOLOv3#4]
hi, I didn't found save_weights function in this code. how can I save the param and test result, thx...

**Comment by catalinolaru1** [comment]
Here, this should solve it. If there are any problems, let me know.

`"""
    @:param path    - path of the new weights file
    @:param cutoff  - save layers between 0 and cutoff (cutoff = -1 -> all are saved)
    """
    def save_weights(self, path, cutoff=-1):

        if cutoff:
            num_layers = len(self.module_list)
        else:
            num_layers = cutoff

        fp = open(path, 'wb')
        self.header_info[3] = self.seen
        self.header_info.tofile(fp)

        for i, (module_def, module) in enumerate(zip(self.module_defs[:cutoff], self.module_list[:cutoff])):
            if module_def['type'] == 'convolutional':
                conv_layer  = module[0]

                # If batch norm, load bn first
                if module_def['batch_normalize']:
                    bn_layer    = module[1]

                    bn_layer.bias.data.cpu().numpy().tofile(fp)
                    bn_layer.weight.data.cpu().numpy().tofile(fp)
                    bn_layer.running_mean.data.cpu().numpy().tofile(fp)
                    bn_layer.running_var.data.cpu().numpy().tofile(fp)
                # Load conv bias
                else:
                    conv_layer.bias.data.cpu().numpy().tofile(fp)

                # Load conv weights
                conv_layer.weight.data.cpu().numpy().tofile(fp)

        fp.close()`

**Comment by eriklindernoren** [comment]
Thanks @catalinolaru1. Feel free to make a PR. :) If not I'll look into adding this.

**Comment by catalinolaru1** [comment]
Sure, i'll make a PR now.

**Comment by SidHard** [comment]
thanks, @catalinolaru1 , I'll try it
